openstack_controller/datadog_checks/openstack_controller/components/swift.py

```diff
@@ -0,0 +1,51 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+
+from datadog_checks.openstack_controller.components.component import Component
+from datadog_checks.openstack_controller.metrics import (
+    SWIFT_CONTAINER_COUNT,
+    SWIFT_CONTAINER_METRICS,
+    SWIFT_CONTAINER_PREFIX,
+    SWIFT_CONTAINER_TAGS,
+    SWIFT_RESPONSE_TIME,
+    SWIFT_SERVICE_CHECK,
+    get_metrics_and_tags,
+)
+
+
+class Swift(Component):
+    ID = Component.Id.SWIFT
+    TYPES = Component.Types.SWIFT
+    SERVICE_CHECK = SWIFT_SERVICE_CHECK
+
+    def __init__(self, check):
+        super(Swift, self).__init__(check)
+
+    @Component.register_global_metrics(ID)
+    @Component.http_error(report_service_check=True)
+    def _report_response_time(self, global_components_config, tags):
+        self.check.log.debug("reporting `%s` response time", Swift.ID.value)
+        response_time = self.check.api.get_response_time(Swift.TYPES.value, remove_project_id=False)
```

Comment:
Can you add a comment describing why we don't want to remove the project ID? And why is this keeping the project ID but being called when collecting global metrics?

Reply:
The endpoint that is being used for Swift is `/v1/{account}`. If Swift were like other components, we could request to `/v1` to get the response time. However, that endpoint doesn't exist, so we need to request `/v1/{account}`, and `{account}` is of the format `'AUTH'+{project_id}`. As such, to get the URL, we still need to keep the project id, even though we are collecting a global metric like response time.

Reply:
Thanks for the explanation, that makes sense. But in this case are we only collecting the response time for one project/account? Do you think it's ever the case where the response time could be drastically different for each account?

Reply:
Different accounts could have differing numbers of containers/objects--however the response times shouldn't be too dissimilar, as most of the response time should be taken by reaching the endpoint (which should be common amongst accounts), as opposed to gathering data at the endpoint.